### PR TITLE
Make changes to THEANO_FLAGS

### LINKED ISSUE #1756: Tests for the sqlite backend fail if floatX=float32
The type conversion from numpy to sqlite seems to break for single precision floats:

Put `floatX = float32` into .theanorc or THEANO_FLAGS and run the sqlite tests:

`nosetests nosetests pymc3/tests/test_sqlite_backend.py`

```
======================================================================
FAIL: test_standard_close (pymc3.tests.test_sqlite_backend.TestSQlite1dSampling)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/adr/git/pymc3/pymc3/tests/backend_fixtures.py", line 237, in test_standard_close
    np.zeros(self.strace.var_shapes[varname]))
  File "/usr/lib/python3.6/site-packages/numpy/testing/utils.py", line 322, in assert_equal
    return assert_array_equal(actual, desired, err_msg, verbose)
  File "/usr/lib/python3.6/site-packages/numpy/testing/utils.py", line 813, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/usr/lib/python3.6/site-packages/numpy/testing/utils.py", line 739, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 100.0%)
 x: array([b'', b''], 
      dtype='|S4')
 y: array([ 0.,  0.]) 
```

### PR DESCRIPTION
**Pull request by souravsingh**
Fixes #1756

### CODE REVIEW

**Comment by souravsingh** [comment]
@twiecki There are some tests failing due to a ``float64`` dtype. Is there a way to fix them?

File: scripts/test.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-THEANO_FLAGS='gcc.cxxflags="-march=core2"' nosetests "$@"
+THEANO_FLAGS='gcc.cxxflags="-march=core2", floatX=float32' nosetests "$@"
```
Comment (twiecki) [inline scripts/test.sh:5]: We should run the tests with both, float32 and float64.

**Comment by twiecki** [comment]
@souravsingh These are important to fix but only way is to go through and fix each one individually. These are actual bugs we haven't revealed before.

If you could help with that it'd be a great contribution (and you'll learn a lot about the code). You can check this PR https://github.com/pymc-devs/pymc3/pull/1690 where I started to do it. You may want to set `warn_float64 = pdb`and then see where the problem occurs.

**Comment by springcoil** [comment]
I'm trying to run this locally with the theano flags inside the docker instance. I get 'warn_float64' not recognized by theano flags as an error. Does anyone have the theano flags to hand?
```/opt/conda/lib/python3.5/site-packages/theano/configdefaults.py:1800: UserWarning: Theano does not recognise this flag: config.warn_float64```

**Comment by springcoil** [comment]
Why did you revert these changes?

**Comment by souravsingh** [comment]
From @ColCarroll comment, I understood that the ``float_64`` settings is not needed, so I reverted the commit. I am sorry if I misunderstood the comment.

File: .travis.yml [hunk 2]
```diff
@@ -19,6 +19,8 @@ env:
   - PYTHON_VERSION=3.6 TESTCMD=" -vv --with-timer --with-coverage --cover-package=pymc3 -e test_examples -e test_distributions"
   - PYTHON_VERSION=3.6 TESTCMD=" -vv --with-timer --with-coverage --cover-package=pymc3 pymc3.tests.test_distributions pymc3.tests.test_examples"
   - PYTHON_VERSION=3.6 TESTCMD=" -vv --with-timer --with-coverage --cover-package=pymc3 pymc3.tests.test_distributions_random"
+  - THEANO_FLAGS='floatX=float32'
```
Comment (twiecki) [inline .travis.yml]: We need all the above test runs, and add a new flag, e.g. THEANO_FLOAT_TYPE. Then in test.sh we need to fish out and add it to THEANO_FLAGS.

Comment (souravsingh) [reply .travis.yml]: Would the variable be some thing like this- `THEANO_FLOAT_TYPE="floatX=float32"`?

Comment (twiecki) [reply .travis.yml]: Yes. It requires some knowledge of bash.

**Comment by souravsingh** [comment]
@twiecki Seven tests are failing due to the float32 precision. Is there a way to fix those?

**Comment by souravsingh** [comment]
Apologies. I clicked on close PR by mistake.

**Comment by twiecki** [comment]
Closing due to inactivity. I think a bit more bash-foo is required here.

**Pull request closed without merging**